pkg/daemon/ceph/client/pool.go

```diff
@@ -183,11 +183,24 @@ func CreatePool(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec
 }
 
 func CreatePoolWithPGs(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, appName, pgCount string) error {
+	autoscaleMode := pool.AutoscaleMode
 	if pool.Name == "" {
 		return errors.New("pool name must be specified")
 	}
+	if pool.Parameters != nil {
+		placementGroupCount, ok := pool.Parameters["pg_num"]
+		if ok {
+			pgCount = placementGroupCount
+			if autoscaleMode == cephv1.PoolAutoscaleModeOn {
+				logger.Warning("The PGs will be managed by the Ceph PG autoscaler, because of the autoscale mode is on, the settings of the PGs is invalid")
```

Comment:
Let's be a bit more clear with this error message so the user knows what two values (from the view they have of the configuration) are in conflict and that they need to change one.

```suggestion
				return fmt.Errorf("the pg_num parameter is invalid with autoscaleMode=%q; one of the values must be corrected", cephv1.PoolAutoscaleModeOn)
```